jax/api.py

```diff
@@ -588,12 +595,47 @@ def vmap(fun, in_axes=0, out_axes=0):
   >>> mv = vmap(vv, (0, None), 0)      #  ([b,a], [a]) -> [b]      (b is the mapped axis)
   >>> mm = vmap(mv, (None, 1), 1)      #  ([b,a], [a,c]) -> [b,c]  (c is the mapped axis)
 
-  Here we use `[a,b]` to indicate an array with shape (a,b). Here are some
+  Here we use ``[a,b]`` to indicate an array with shape (a,b). Here are some
   variants:
 
   >>> mv1 = vmap(vv, (0, 0), 0)   #  ([b,a], [b,a]) -> [b]        (b is the mapped axis)
   >>> mv2 = vmap(vv, (0, 1), 0)   #  ([b,a], [a,b]) -> [b]        (b is the mapped axis)
   >>> mm2 = vmap(mv2, (1, 1), 0)  #  ([b,c,a], [a,c,b]) -> [c,b]  (c is the mapped axis)
+
+  Here's an example of using container types in ``in_axes`` to specify which
+  axes of the container elements to map over:
+
+  >>> A, B, C, D = 2, 3, 4, 5
+  >>> x = np.ones((A, B))
+  >>> y = np.ones((B, C))
+  >>> z = np.ones((C, D))
+  >>> def foo(tree_arg):
+  ...   x, (y, z) = tree_arg
+  ...   return np.dot(x, np.dot(y, z))
+  >>> tree = (x, (y, z))
+  >>> print(foo(tree))
+  [[12. 12. 12. 12. 12.]
+   [12. 12. 12. 12. 12.]]
+  >>> from jax import vmap
+  >>> K = 6  # batch size
+  >>> x = np.ones((K, A, B))  # batch axis in different locations
+  >>> y = np.ones((B, K, C))
+  >>> z = np.ones((C, D, K))
+  >>> tree = (x, (y, z))
+  >>> vfoo = vmap(foo, in_axes=((0, (1, 2)),))
+  >>> print(vfoo(tree))
+  [[[12. 12. 12. 12. 12.]
+    [12. 12. 12. 12. 12.]]
+   [[12. 12. 12. 12. 12.]
+    [12. 12. 12. 12. 12.]]
+   [[12. 12. 12. 12. 12.]
+    [12. 12. 12. 12. 12.]]
+   [[12. 12. 12. 12. 12.]
+    [12. 12. 12. 12. 12.]]
+   [[12. 12. 12. 12. 12.]
+    [12. 12. 12. 12. 12.]]
+   [[12. 12. 12. 12. 12.]
+    [12. 12. 12. 12. 12.]]]
```

Comment:
Nice example! Maybe print vfoo(tree).shape for brevity?

Reply:
Done.